pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -1642,6 +1647,27 @@ func (tmpl *Template) GetType() TemplateType {
 	return TemplateTypeUnknown
 }
 
+// GetNodeType returns the type of this template as a Node Type
+//
+func (tmpl *Template) GetNodeType() NodeType {
+	if tmpl.Container != nil {
+		return NodeTypePod
+	}
+	if tmpl.Steps != nil {
+		return NodeTypeSteps
+	}
+	if tmpl.DAG != nil {
+		return NodeTypeDAG
+	}
+	if tmpl.Script != nil {
+		return NodeTypePod
+	}
+	if tmpl.Suspend != nil {
+		return NodeTypeSuspend
+	}
+	return NodeTypePod
```

Comment:
Not sure NodeTypePod should be the default here

Reply:
It should not, the default should be a panic, since at this point we have considered all possible `NodeTypes`